rules-java-archives/addon/src/main/java/org/jboss/windup/rules/apps/java/archives/freemarker/GetArchivesBySHA1Method.java

```diff
@@ -30,7 +35,18 @@ public Object exec(List arguments) throws TemplateModelException
         SimpleScalar freemarkerArg = (SimpleScalar) arguments.get(0);
         String sha1 = freemarkerArg.getAsString();
         ArchiveService archiveService = new ArchiveService(context);
-        return archiveService.findBySHA1(sha1);
+
+        // filter out the one in the shared-libs virtual app.
+        Predicate<ArchiveModel> predicate = new Predicate<ArchiveModel>()
+        {
+            @Override
+            public boolean apply(ArchiveModel input)
+            {
+                return !input.getDuplicateArchives().iterator().hasNext();
```

Comment:
Strange way to do this, it could be changed in the future. (Jess said this himself :)